cppsdk/include/playfab/PlayFabEvent.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#ifndef DISABLE_PLAYFABENTITY_API
+
+#include <playfab/PlayFabError.h>
+#include <playfab/PlayFabEventsDataModels.h>
+
+namespace PlayFab
+{
+    enum struct PlayFabEventServerProcessingType
+    {
+        Default,
+        Lightweight,
+        Heavyweight
+    };
+
+    /// <summary>
+    /// Interface for any event
+    /// </summary>
+    struct IPlayFabEvent
```

Comment:
>struct [](start = 4, length = 6)

Did you have any thoughts on making these structs vs class ? In my mind I usually use struct for simple data type aggregates. Curious to hear your reasoning here :)

Reply:
Exactly the same reasoning as you just said. They are meant to be just data holders without any business logic (beside possible convenience constructors and/or getters/setters). I of course don't have a strong preference and more than willing to rename them into classes. My intent was just to show to customer that it is meant to be just data.

---
In reply to: [227944488](https://github.com/PlayFab/XPlatCppSdk/pull/6#discussion_r227944488) [](ancestors = 227944488)